[seal] Allow disabling SEAL_THROW_ON_TRANSPARENT_CIPHERTEXT

kiromaru opened the pull request:
Allow disabling the option SEAL_THROW_ON_TRANSPARENT_CIPHERTEXT. Some
third party projects depend on this option being disabled.

- What does your PR fix? Allows disabling a configuration option for SEAL

- Which triplets are supported/not supported? Have you updated the CI baseline?
Tested x64-windows-static, x64-linux, x64-osx.
SEAL only supports static linkage in Windows, updated the CI baseline to reflect this.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes

kiromaru committed:
Allow disabling SEAL_THROW_ON_TRANSPARENT_CIPHERTEXT

Allow disabling the option SEAL_THROW_ON_TRANSPARENT_CIPHERTEXT. Some
third party libraries depend on this option being disabled.

kiromaru committed:
Udpate CI baseline

SEAL only supports static linkage. Builds will fail if dependencies are
installed before installing SEAL. CI baseline nees to be updated to
reflect builds that we expect to fail.

JackBoosY commented on scripts/ci.baseline.txt:
```diff
@@ -1380,7 +1380,10 @@ sdl2-net:arm-uwp=fail
 sdl2-net:x64-uwp=fail
 # https://github.com/microsoft/vcpkg/issues/10918
 seal:arm-uwp=fail
+seal:arm64-windows=fail
 seal:x64-uwp=fail
+seal:x64-windows=fail
+seal:x86-windows=fail
```
Why add these 3 lines?

kiromaru replied:
The SEAL library works only with static linkage in Windows. The CMake file makes sure of this, and will generate an error if it detects that dependencies are not available as static. This check was not working correctly before, the library would compile but would fail when trying to use it in a project. We pushed a hot-fix to GitHub to fix this, and the pull request reflects this new commit. As a result builds fail correctly now when non-static linkage is detected, so the failures in those three triplets are expected.

JackBoosY replied:
If this failure is by design, please add `Support: !static` to `CONTROL` file instead of adding this.

JackBoosY replied:
And please bump the port version. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

kiromaru replied:
I've bumped the port version and added the "Supports" line to `CONTROL`. I reverted the changes to ci.baseline.txt as well.

kiromaru committed:
Use vcpkg.json instead of CONTROL

This allows us to use the "port-version" to update SEAL port with the same
 source version.

kiromaru committed:
Revert "Use vcpkg.json instead of CONTROL"

This reverts commit 2f8a081dabf4299d6fcc2c0e11a6cb67a68ff3e5.

kiromaru committed:
Modify CONTROL file

Update port version and indicate supported triplets

kiromaru committed:
Fix supports expression

kiromaru commented:
Thank you for the review, we will actually update the version of SEAL to 3.6.1 which has just been released and include the change there.
